pkg/cri/resource-manager/policy/builtin/balloons/cputree_test.go

```diff
@@ -0,0 +1,606 @@
+						threadTree.AddCpus(cpuset.NewCPUSet(cpuID))
+						csit[cpuID] = cpuInTopology{
+							packageID, dieID, numaID, coreID, threadID, cpuID,
+							packageTree.name, dieTree.name, numaTree.name, coreTree.name, threadTree.name,
+							fmt.Sprintf("cpu%d", cpuID),
+						}
+						cpuID += 1
+					}
+				}
+			}
+		}
+	}
+	return sysTree, csit
+}
+
+func verifyNotOn(t *testing.T, nameContents string, cpus cpuset.CPUSet, csit cpusInTopology) {
+	for _, cpuID := range cpus.ToSlice() {
+		name := csit[cpuID].threadName
+		if strings.Contains(name, nameContents) {
+			t.Errorf("cpu%d (%s) in unexpected region %s", cpuID, name, nameContents)
+		}
+	}
+}
+
+func verifySame(t *testing.T, topoLevel string, cpus cpuset.CPUSet, csit cpusInTopology) {
+	seenName := ""
+	seenCpuID := -1
+	for _, cpuID := range cpus.ToSlice() {
+		cit := csit[cpuID]
+		thisName := cit.TopoName(topoLevel)
+		thisCpuID := cit.cpuID
+		if thisName == "" {
+			t.Errorf("unexpected (invalid) topology level %q", topoLevel)
+		}
+		if seenName == "" {
+			seenName = thisName
+			seenCpuID = cit.cpuID
+		}
+		if seenName != thisName {
+			t.Errorf("expected the same %s, got: cpu%d in %s, cpu%d in %s",
+				topoLevel,
+				seenCpuID, seenName,
+				thisCpuID, thisName)
+		}
+	}
+}
+
+func (csit cpusInTopology) getElements(topoLevel string, cpus cpuset.CPUSet) []string {
+	elts := []string{}
+	for _, cpuID := range cpus.ToSlice() {
+		elts = append(elts, csit[cpuID].TopoName(topoLevel))
+	}
+	return elts
+}
+
+func (csit cpusInTopology) verifyDisjoint(t *testing.T, topoLevel string, cpusA cpuset.CPUSet, cpusB cpuset.CPUSet) {
+	eltsA := csit.getElements(topoLevel, cpusA)
+	eltsB := csit.getElements(topoLevel, cpusB)
+	for _, eltA := range eltsA {
+		for _, eltB := range eltsB {
+			if eltA == eltB {
+				t.Errorf("expected disjoint %ss, got %s on both cpusets %s and %s",
+					topoLevel, eltA, cpusA, cpusB)
+				return
+			}
+		}
+	}
+}
+
+/*
+topology: [5]int{2, 2, 2, 2, 2},
+allocations: []int{
+	0,  // cpu on p0/d0/n0/c0/t0
+	1,  // cpu on p0/d0/n0/c0/t1
+	2,  // cpu on p0/d0/n0/c1/t0
+	3,  // cpu on p0/d0/n0/c1/t1
+	4,  // cpu on p0/d0/n1/c0/t0
+	5,  // cpu on p0/d0/n1/c0/t1
+	6,  // cpu on p0/d0/n1/c1/t0
+	7,  // cpu on p0/d0/n1/c1/t1
+	8,  // cpu on p0/d1/n0/c0/t0
+	9,  // cpu on p0/d1/n0/c0/t1
+	10, // cpu on p0/d1/n0/c1/t0
+	11, // cpu on p0/d1/n0/c1/t1
+	12, // cpu on p0/d1/n1/c0/t0
+	13, // cpu on p0/d1/n1/c0/t1
+	14, // cpu on p0/d1/n1/c1/t0
+	15, // cpu on p0/d1/n1/c1/t1
+	16, // cpu on p1/d0/n0/c0/t0
+	17, // cpu on p1/d0/n0/c0/t1
+	18, // cpu on p1/d0/n0/c1/t0
+	19, // cpu on p1/d0/n0/c1/t1
+	20, // cpu on p1/d0/n1/c0/t0
+	21, // cpu on p1/d0/n1/c0/t1
+	22, // cpu on p1/d0/n1/c1/t0
+	23, // cpu on p1/d0/n1/c1/t1
+	24, // cpu on p1/d1/n0/c0/t0
+	25, // cpu on p1/d1/n0/c0/t1
+	26, // cpu on p1/d1/n0/c1/t0
+	27, // cpu on p1/d1/n0/c1/t1
+	28, // cpu on p1/d1/n1/c0/t0
+	29, // cpu on p1/d1/n1/c0/t1
+	30, // cpu on p1/d1/n1/c1/t0
+	31, // cpu on p1/d1/n1/c1/t1
+},
+*/
```

Comment:
Is there a reason to have this commented-out block?

Reply:
Pondered between removing this section or commenting why it is there.

Now ended up adding a comment to give a reason why it exists. If you prefer, I can also delete it. However, it may increase the pain of the poor developer who has to debug why a test starts failing, or write a test. In the first glance it is not obvious where in the topology CPU _i_ belongs to. And I thought that an explicit list might be easier for a developer who doesn't notice that one can get 2-2-2-2-2-topology location for CPU _i_ with

```
(echo obase=2; echo i) | bc
```
or
```
python3 -c 'print(bin(i))'
```
etc.